### DISCUSSION TITLE
send data from micropython to raspberry

### DISCUSSION THREAD

**Discussion by ewok2** [micropython/micropython#12848] [Using MicroPython] [answered]
Hello
I am a newbi in micropython. All my experience was with python script running on raspberry...
My hardware board is a raspberry pi pico W

I need to send data collect from the micropython board to a share memory on a raspberry.
Between raspberry I use pymemcache library but it seem's this library is not available with "mip.install" on micropyton.

What could you sugest to achive this data transfert?
- install the pymemcache with another methods?
- use a micropython built in methods to send data through neworks?

Thanks for help

**Comment by sophiedegran** [comment]
Hello,
I don't know if the 2 boards are on same lan and the size you wan't to transfer.. you can simply use a client ftp on micropython and a server ftp on your raspberry ..

**Reply by ewok2** [reply]
Yes same LAN, and only some values to push.
For exemple I have to push T° and Humidity value each xx minutes.
On the other side I have a prometheus exporter and I get theses value to display and record them with Grafana.

The ftp is indeed an idee (throught a file with the data inside)
I am also looking to a MQTT server side (but it seem's eavy only for 6 number values to update)

I have try to put the pymemecache lib on the Pico. I have to add other lib that are import by pymemcache and at the end I need the "Typing" lib that seem's not exist for micropython...

**Reply by sophiedegran** [reply]
Yes you're right ... you can also send by websocket (they are so many solutions!).. MQTT for me is good but so heavy for that
Have good exploration ☺

**Reply by shariltumin** [reply, marked as answer]
Or you can write a simple TCP/IP server on the Raspberry Pi and a simple TCP/IP client on the MicroPython wifi-enabled boards using python socket library.

**Reply by ewok2** [reply]
thanks I will have a look on the TCP/IP solution with python socket library

**Reply by mendenm** [reply]
is this a case where you can't just  use a USB cable, and send data over the serial console to the regular RPi?  You didn't clarify the details of your setup.  If not, the 'microdot' web server runs very easily on the Pico-W, and takes only a few minutes to set up a data server.

**Reply by ewok2** [reply]
Oups indeed I did not explain all...

Initialy I had a raspberry pi zero supply by solar panel and cell during the night.
This raspberry was running script that get somme data outside of my house and is not connected by usb or ethernet to the other server / raspberry of my network.
But the raspberry can't pass in sleep mode and need to much power. I can't pass 24h when winter is comming...

So I decide to use a Pico and I steel need to put some data over wifi.
More clear ? ;-)

**Comment by ewok2** [comment]
Thanks all
A socket client sever is now working fine.
I pass a dictionary with "str(dict)" from the client and get it again with "ast.literla_eval(message)" to the server
